homeassistant/components/sensor/apcupsd.py

```diff
@@ -11,33 +11,109 @@
+    'mintimel': ['Shutdown Time', '', 'mdi:timer'],
+    'model': ['Model', '', 'mdi:information-outline'],
+    'nombattv': ['Battery Nominal Voltage', 'V', 'mdi:flash'],
+    'nominv': ['Nominal Input Voltage', 'V', 'mdi:flash'],
+    'nomoutv': ['Nominal Output Voltage', 'V', 'mdi:flash'],
+    'nompower': ['Nominal Output Power', 'W', 'mdi:flash'],
+    'numxfers': ['Transfer Count', '', 'mdi:counter'],
+    'outputv': ['Output Voltage', 'V', 'mdi:flash'],
+    'reg1': ['Register 1 Fault', '', 'mdi:information-outline'],
+    'reg2': ['Register 2 Fault', '', 'mdi:information-outline'],
+    'reg3': ['Register 3 Fault', '', 'mdi:information-outline'],
+    'retpct': ['Restore Requirement', '%', 'mdi:battery-alert'],
+    'selftest': ['Last Self Test', '', 'mdi:calendar-clock'],
+    'sense': ['Sensitivity', '', 'mdi:information-outline'],
+    'serialno': ['Serial Number', '', 'mdi:information-outline'],
+    'starttime': ['Startup Time', '', 'mdi:calendar-clock'],
+    'statflag': ['Status Flag', '', 'mdi:information-outline'],
+    'status': ['Status', '', 'mdi:information-outline'],
+    'stesti': ['Self Test Interval', '', 'mdi:information-outline'],
+    'timeleft': ['Time Left', '', 'mdi:clock-alert'],
+    'tonbatt': ['Time on Battery', '', 'mdi:timer'],
+    'upsmode': ['Mode', '', 'mdi:information-outline'],
+    'upsname': ['Name', '', 'mdi:information-outline'],
+    'version': ['Daemon Info', '', 'mdi:information-outline'],
+    'xoffbat': ['Transfer from Battery', '', 'mdi:transfer'],
+    'xoffbatt': ['Transfer from Battery', '', 'mdi:transfer'],
+    'xonbatt': ['Transfer to Battery', '', 'mdi:transfer'],
+}
+
 SPECIFIC_UNITS = {
-    "ITEMP": TEMP_CELSIUS
+    'ITEMP': TEMP_CELSIUS
+}
+INFERRED_UNITS = {
+    ' Minutes': 'min',
+    ' Seconds': 'sec',
+    ' Percent': '%',
+    ' Volts': 'V',
+    ' Watts': 'W',
+    ' Hz': 'Hz',
+    ' C': TEMP_CELSIUS,
 }
 
 _LOGGER = logging.getLogger(__name__)
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Setup the APCUPSd sensor."""
-    typ = config.get(apcupsd.CONF_TYPE)
-    if typ is None:
-        _LOGGER.error(
-            "You must include a '%s' when configuring an APCUPSd sensor.",
-            apcupsd.CONF_TYPE)
-        return False
-    typ = typ.upper()
-
-    if typ not in apcupsd.DATA.status:
-        _LOGGER.error(
-            "Specified '%s' of '%s' does not appear in the APCUPSd status "
-            "output.", apcupsd.CONF_TYPE, typ)
-        return False
-
-    add_entities((
-        Sensor(config, apcupsd.DATA, unit=SPECIFIC_UNITS.get(typ)),
-    ))
+    """Setup the APCUPSd sensors."""
+    entities = []
+
+    for resource in config['resources']:
+        sensor_type = resource.lower()
+
+        if sensor_type not in SENSOR_TYPES:
+            SENSOR_TYPES[sensor_type] = [
+                sensor_type.title(), '', 'mdi:mdi:information-outline']
+
+        if sensor_type.upper() not in apcupsd.DATA.status:
```

Comment:
I had two items in my `apcaccess` output that were not documented: `nominv` and `driver`

Because of this, I do not want to have to release an update for every time someone finds an undocumented value supplied by the UPS.